nav2-usability/scenario-code/dynamic-linking/router.dart

```diff
@@ -0,0 +1,247 @@
+// Copyright 2021, the Flutter project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:math';
+
+import 'package:flutter/material.dart';
+
+void main() {
+  runApp(WishlistApp());
+}
+
+class Wishlist {
+  final String id;
+
+  Wishlist(this.id);
+}
+
+class AppState extends ChangeNotifier {
+  final List<Wishlist> wishlists = <Wishlist>[];
+  Wishlist? _selectedWishlist;
+
+  Wishlist? get selectedWishlist => _selectedWishlist;
+
+  set selectedWishlist(Wishlist? w) {
+    _selectedWishlist = w;
+    notifyListeners();
+  }
+
+  void addWishlist(Wishlist wishlist) {
+    wishlists.add(wishlist);
+    notifyListeners();
+  }
+}
+
+class WishlistApp extends StatefulWidget {
+  @override
+  _WishlistAppState createState() => _WishlistAppState();
+}
+
+class _WishlistAppState extends State<WishlistApp> {
+  final BookRouterDelegate _routerDelegate = BookRouterDelegate();
```

Comment:
_WishlistAppState should dispose _routerDelegate in its dispose method

Reply:
Good catch, that's fixed in all three samples now.